iocore/net/quic/QUICAckFrameCreator.h

```diff
@@ -27,35 +27,66 @@
 #include "QUICFrameGenerator.h"
 #include "QUICTypes.h"
 #include "QUICFrame.h"
-#include <vector>
-#include <set>
+#include <list>
 
-class QUICAckPacketNumbers
+class QUICAckFrameCreator : public QUICFrameGenerator
 {
 public:
-  void push_back(QUICPacketNumber packet_number);
-  QUICPacketNumber front();
-  QUICPacketNumber back();
-  size_t size();
-  void clear();
-  void sort();
+  class QUICAckPacketNumbers : public QUICFrameGenerator
```

Comment:
Not sure here. I can change it if you don't like.

Reply:
A reason I'm not so keen to make QUICAckPacketNumbers inherit FrameGenerator is that I don't see reasons to do this. What was the motivation of moving `generate_frame` logic into QUICAckPacketNumbers?

It seems like `QUICAckPacketNumbers` does most things and QUICAckFrameCreator just checks pre-conditions.

If there's no reasons, I'd suggest:
- Keep QUICAckPacketNumbers as data container
- Generate frames in QUICAckFrameCreator

Then, QUICAckFrameCreator knows how to create frames and is responsible for creating frames base on received packet information, and QUICAckPacketNumbers is responsible for maintain the information such as numbers, size, last_ack_number, largest_ack_received_time, etc. Even if we changed how we create frames, what we maintain the information wouldn't be changed, and vice versa.

And we should rename classes because they have size and ack_only flag:
- QUICAckPacketNumbers -> ReceivedPacketRecords
- Rename RecvPacketNumber -> PacketRecord